code/modules/vehicles/mecha/mecha_mob_interaction.dm

```diff
@@ -184,9 +192,15 @@
 /obj/vehicle/sealed/mecha/container_resist_act(mob/living/user)
 	if(isAI(user))
 		var/mob/living/silicon/ai/AI = user
-		if(!AI.can_shunt)
-			to_chat(AI, span_notice("You can't leave a mech after dominating it!."))
-			return FALSE
+		if(!AI.linked_core)
+			to_chat(AI, span_userdanger("Inactive core destroyed. Unable to return."))
+			if(!AI.can_shunt || !AI.hacked_apcs.len)
+				to_chat(AI, span_warning(AI.can_shunt ? "No hacked APCs available." : "No shunting capabilities."))
+				return FALSE
+			var/confirm = tgui_alert(AI, "Shunt to a random APC? You won't have anywhere else to go!", "Confirm Emergency Shunt", list("Yes", "No"))
+			if(confirm == "Yes")
```

Comment:
This should have a sanity check that the AI is still in a valid state to shunt. Primarily `is the ai still actually in the mech?`, but other things like `are they dead` may be valid, idk when this is invoked.